### ISSUE TITLE
tpm2_nvdefine without specifying the size of data area got "structure is the wrong size(0x2D5)"

### ISSUE THREAD

**Issue by Ivanhu5866** [tpm2-software/tpm2-tools#2171]
I tried to test tpm2_nvdefine without specifying the size of data area with several latest machines, and lots of machines got "the structure is the wrong size(0x2D5)" error.
```
$sudo tpm2_nvdefine
WARNING:esys:src/tss2-esys/api/Esys_NV_DefineSpace.c:333:Esys_NV_DefineSpace_Finish() Received TPM Error
ERROR:esys:src/tss2-esys/api/Esys_NV_DefineSpace.c:122:Esys_NV_DefineSpace() Esys Finish ErrorCode (0x000002d5)
ERROR: Failed to define NV area at index 0x1000000
ERROR: Esys_NV_DefineSpace(0x2D5) - tpm:parameter(2):structure is the wrong size
ERROR: Unable to run tpm2_nvdefine
```
I believe this is because the size defaults to MAX_NV_INDEX_SIZE which is typically 2048.
But under these machines, the value TPM2_PT_NV_INDEX_MAX is 1664(0x680). 
$ sudo tpm2_getcap properties-fixed
...
TPM2_PT_NV_INDEX_MAX:
  raw: 0x680
...
I also tried to specify the size with -s, when over 0x680, then got the same error returned.

Tpm2-tools version is 4.1.1
Ubuntu Focal(20.04)

**Comment by williamcroberts** [comment]
@idesai this would be a good one to add to 5.0 for the behavioral changes. @Ivanhu5866 has a patch that perhaps he can share. I see a patch referenced at https://github.com/Ivanhu5866/tpm2-tools/commit/785cc544b30bb38cdd01917a86119409f474fcb9

**Comment by idesai** [comment]
fixed via #2173
